File: codeforces/o.cpp
#include <unordered_map>
#include <vector>
#include <stdio.h>
#include <stdlib.h>
#include <algorithm>

using namespace std;
typedef long long ll;

int main() {
  int k, n;
  scanf("%d\n%d", &k, &n);

  vector<vector<int>> mapper(k + 1);
  vector<int> values;

  int in_class = 0;
  int all_class = 0;
  for (int i = 0; i != n; ++i) {
    int curr_class, value;
    scanf("%d%d", &value, &curr_class);

    values.push_back(value);
    mapper[curr_class].push_back(value);
    printf("%d\n", mapper[curr_class].size());
  }

  printf("Here!");
  sort(values.begin(), values.end());
  for (int i = 0; i != n; ++i) {
    sort(mapper[i].begin(), mapper[i].end());
  }

  for (int i = 1; i != n; ++i) {
    all_class += ((ll) 1) * i * (n - i) * (values[i] - values[i - 1]);
  }  

  for (int kk = 0; kk != k; ++k) {
    for (int i = 0; i != mapper[kk].size(); ++i) {
      in_class += ((ll) 1) * i * (n - i) * (mapper[kk][i] - mapper[kk][i - 1]);
    }
  }

  printf("%d\n%d", in_class, all_class - in_class);
  return 0;
}

File: codeforces/o.py

k, n = int(input()), int(input())

mapper = [[] for _ in range(k + 1)]
values = [0] * n

in_class = 0
all_class = 0
for i in range(n):
  value, curr_class = list(map(lambda x : int(x), input().split()))

  values[i] = value
  mapper[curr_class].append(value)

values.sort()
for i in range(1, k + 1):
  mapper[i].sort()

for i in range(1, n):
  all_class += i * (n - i) * (values[i] - values[i - 1])

for kk in range(1, k + 1):
  nn = len(mapper[kk])
  for i in range(1, nn):
    in_class += i * (nn - i) * (mapper[kk][i] - mapper[kk][i - 1])

in_class *= 2
all_class *= 2

print(in_class)
print(all_class - in_class)